engine/org.eclipse.birt.report.engine.emitter.html/src/org/eclipse/birt/report/engine/emitter/html/HTMLPerformanceOptimize.java

```
@@ -545,16 +547,58 @@ public void buildImageStyle(IImageContent image, StringBuffer styleBuffer, int d
-			imgHeight = new DimensionType(image.getImageCalculatedSize().getHeight(), "px");
+			imgHeight = new DimensionType(image.getImageCalculatedSize().getHeight(), DimensionType.UNITS_PX);
+		}
+
+		boolean useHeightAutoValue = false;
+		try {
+			// column width
+			DimensionType columnWidth = ((ICellContent) image.getParent()).getColumnInstance().getWidth();
+			if (image.isFitToContainer() && columnWidth != null) {
+
+				// convert column with "px" to "mm"
+				DimensionType compareColumnWidth = new DimensionType(columnWidth.getMeasure(), columnWidth.getUnits());
+				if (columnWidth.getUnits().toLowerCase().equals(DimensionType.UNITS_PX)) {
+					double widthMM = (columnWidth.getMeasure() * 25.4) / 96;
+					compareColumnWidth = new DimensionType(widthMM, DimensionType.UNITS_MM);
+				}
+				DimensionValue compareColumnWidthValue = DimensionUtil.convertTo(compareColumnWidth.getMeasure(),
+						compareColumnWidth.getUnits(), DimensionType.UNITS_MM);
+
+				// image width raw size
+				if (imgWidth == null) {
+					imgWidth = new DimensionType(image.getImageCalculatedSize().getWidth(), DimensionType.UNITS_PX);
+				}
+				DimensionType compareImageWidth = new DimensionType(imgWidth.getMeasure(), imgWidth.getUnits());
+
+				// convert image width "px" to "mm"
+				if (imgWidth.getUnits().toLowerCase().equals(DimensionType.UNITS_PX)) {
+					double widthMM = (imgWidth.getMeasure() * 25.4) / 96;
+					compareImageWidth = new DimensionType(widthMM, DimensionType.UNITS_MM);
+				}
+				DimensionValue compareImageWidthValue = DimensionUtil.convertTo(compareImageWidth.getMeasure(),
+						compareImageWidth.getUnits(), DimensionType.UNITS_MM);
+
+				if (compareImageWidthValue.getMeasure() > compareColumnWidthValue.getMeasure()) {
+					imgWidth = new DimensionType(compareColumnWidthValue.getMeasure(), DimensionType.UNITS_MM);
+					useHeightAutoValue = true;
+				}
+			}
+		} catch (Exception e) {
```

I don't like this "silently ignore all errors". 
What exceptions are to be expected here?
It may be ok to silently ignore some expected exceptions, but this is too general.
At least, a message should be logged.

---

Hi both, I confirm, unknown errors are not very friendly. 
The reason of it was that we have many unit-options and it could be that there are some options which can crash the calculation. I will double check for the logging option, this would be better instead of "silent dead".

---

After some additional testing I removed the try-catch completely.